engine/pong.cpp

```diff
@@ -92,85 +99,43 @@ void Pong::Cleanup()
     GetInstance().GetAudioMixer().Cleanup();
 }
 
-void Pong::LoadSceneNext(Scene scene)
+void Pong::LoadSceneNext(SceneType sceneType)
 {
-    GetInstance().mNextScene = scene;
+    GetInstance().mNextScene = sceneType;
     GetInstance().mChangeSceneRequested = true;
 }
 
-void Pong::LoadScene(Scene scene)
+void Pong::LoadScene(SceneType sceneType)
 {
     Pong::Reset();
 
-    switch (scene)
+    const auto LoadSceneObjects = [&](auto scene) {
+        scene.BuildScene();
+        GetInstance().mGameObjects = scene.TransferGameObjects();
+    };
+
+    switch (sceneType)
     {
-        case Scene::TitleScreen:
+        case SceneType::Title:
```

Comment:
Another use for the Visitor perhaps?

Reply:
Yeah thats valid. I think I will keep if for now just because it is a really simple system, and I don't plan on adding more than maybe one more scene type.